jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/DelayedHandler.java

```diff
@@ -123,10 +150,17 @@ protected DelayedProcess newDelayedProcess(boolean contentExpected, String conte
                     yield new UntilMultipartDelayedProcess(handler, request, response, callback, contentType, mpc);
                 yield null;
             }
-            default -> new UntilContentDelayedProcess(handler, request, response, callback);
+            // if other mimeType, then only delay until content if configured
+            default -> delayDispatchUntilContent ? newUntilContentDelayedProcess(handler, request, response, callback) : null;
+
         };
     }
 
+    protected DelayedProcess newUntilContentDelayedProcess(Handler handler, Request request, Response response, Callback callback)
+    {
+        return new UntilContentDelayedProcess(handler, request, response, callback, -1);
```

Comment:
The last parameter `-1` is hardcoded, so applications have no way to specify a value without subclassing, or by specifying a property in the module.

I think it should be a `long` and be configurable in `DelayedHandler` as a setter.
I was thinking that the value `0` would mean "wait until the first chunk", as well as `1`.
Larger values would cap the bytes in the chunk(s).
Then, `-1` could have the meaning of "read it until the last chunk", like form and multipart do.
Default would be `0`, i.e. until first chunk.
I would switch to a `long` because 2 GiB may not that much for specific use cases.

Reply:
I don't mind having a configuration option like that, but I would still like some value that mean 1 input buffer size.  This is a good default value as it means we do not need to configure another value unless we really care and it means zero additional memory commitment from this handler.    I'll have a play